annotator/src/main/java/org/cbioportal/annotator/internal/GenomeNexusImpl.java

```diff
@@ -507,6 +510,25 @@ private String resolveConsequence() {
         }
         return "";
     }
+    
+    private String resolveEntrezGeneId(boolean replace) {
+        if (!replace) {
+            return mRecord.getEntrez_Gene_Id();
+        }
+        // make the rest call to ensembl server for gene external refs
+        RestTemplate restTemplate = new RestTemplate();
+        HttpEntity<LinkedMultiValueMap<String, Object>> requestEntity = getRequestEntity();        
+        ResponseEntity<GeneXref[]> responseEntity = restTemplate.exchange(geneXrefsServiceUrl + canonicalTranscript.getGeneId(), HttpMethod.GET, requestEntity, GeneXref[].class);
```

Comment:
@sheridancbio @zheins right now we are resolving which canonical transcript to use...so this is using the ensembl gene accession (`gene_id`) from the canonical transcript we get/use